client.go

```diff
@@ -209,8 +212,16 @@ func (c *Client) do(ctx context.Context, endpoint *url.URL, apiRequest interface
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("%d: %s", resp.StatusCode, string(respBody))
+		return nil, StatusError{
+			Status:    resp.StatusCode,
+			Message:   string(respBody),
+			Retryable: isRetryable(resp.StatusCode),
```

Comment:
I think the status is enough, to decide if you want to retry or not, so the field `Retryable` seems useless.

Reply:
Fixed it.